src/main.py

```diff
@@ -1110,6 +1141,25 @@ def _get_db_state(self):
         logging.info('Database cluster state is: %s' % state)
         return state
 
+    def _acquire_replication_source_lock(self, primary):
+        if not self.config.getboolean('global', 'use_replication_slots'):
+            return
+        # We need to drop the slot in the old primary.
+        # But we don't know who the primary was (probably there are many of them).
+        # So, we need to release the lock on all hosts.
+        all_hosts = self.zk.get_children(self.zk.MEMBERS_PATH)
```

Comment:
Maybe loop over children of `self.zk.HOST_REPLICATION_SOURCES`?
It has only hosts that ever was a replications source (with locks), it should be faster than all hosts.